find-trio-denovo.py

```diff
@@ -14,167 +13,105 @@
-	thirdCol = sys.argv[4]
+	inFileName = sys.argv[1]
+        momIdx = sys.argv.index("mom")
+        dadIdx = sys.argv.index("dad")
+        childIdx = sys.argv.index("child")
 
 	variant_count = 0 #counts phased and unphased variants
 	unphased_count = 0 #separate counter for variants with all lines treated as unphased
 
-	with open (inFileName,'r') as infile:  #when you use "with open" you don't have to close the file later
-		for line in infile:
-			if line.startswith("#"):   # header and info lines start with "#"
-				recordVariant(line)   
-				unphasedVariants(line)
-			else:  
-				if firstCol == "dad" and secondCol == "mom":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, dadgeno, momgeno, childgeno)= line.strip("\n").split("\t")
-				elif firstCol == "dad" and secondCol == "child":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, dadgeno, childgeno, momgeno)= line.strip("\n").split("\t")
-				elif firstCol == "mom" and secondCol =="dad":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, momgeno, dadgeno, childgeno)= line.strip("\n").split("\t")
-				elif firstCol == "mom" and secondCol == "child":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, momgeno, childgeno, dadgeno)= line.strip("\n").split("\t")
-				elif firstCol == "child" and secondCol == "dad":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, childgeno, dadgeno, momgeno)= line.strip("\n").split("\t")
-				elif firstCol ==  "child" and secondCol == "mom":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, childgeno, momgeno, dadgeno)= line.strip("\n").split("\t")
-
-				# split each of the dad/mom/child columns by ":", to access only the genotype
-				# first element of the list is the genotype when format is Genotype:Quality:ReadDepth:etc.
-				dadgeno_list = dadgeno.split(":")
-				dadGeno = (dadgeno_list[0])
-				
-				momgeno_list = momgeno.split(":")
-				momGeno = (momgeno_list[0])
-				
-				childgeno_list = childgeno.split(":")
-				childGeno = (childgeno_list[0])
-				
-				#now split the genotypes into individual alleles
-				#will split on "/" or "|"
-				(dadAllele1, dadAllele2) = re.split(r"/|\|",dadGeno)	    #to apply to both phased 
-				(momAllele1, momAllele2) = re.split(r"/|\|",momGeno)	    #and unphased entries
-				(childAllele1, childAllele2) = re.split(r"/|\|",childGeno)
-
-				#checks which delimiter was used.  | = phased .   / = unphased
-				#assumes that delimiter on dadGeno is same as momGeno and childGeno 
-				dialect = csv.Sniffer().sniff(dadGeno,["/","|"])
-				delim = (dialect.delimiter)
-
-				#create two lists; one containing both child alleles, the other containing all parent alleles
-				childAlleles = [childAllele1, childAllele2]
-				parentAlleles = [dadAllele1, dadAllele2, momAllele1, momAllele2]
-				
-				child1_in_dad = 0   #to track whether alleles match or not
-				child2_in_dad = 0   
-				child1_in_mom = 0
-				child2_in_mom = 0
```

Comment:
Variables with 1, 2, 3 ... in the names are generally a bad sign. It means you're manually unrolling a loop. Let the computer do the optimizations for you. Keep your code simple by learning to think in arrays. The nice part about an array or list is that they have arbitrary length. You could parse all the data in the samples columns and then later decide not to generate anything from them, but at least you have them in memory should you later want to work with them. You wont have to go back and add `child3_in_data`, `child4_in_dad`, etc.